src/librustc_target/abi/mod.rs

```diff
@@ -871,8 +871,26 @@ impl Niche {
 
 #[derive(PartialEq, Eq, Hash, Debug, HashStable_Generic)]
 pub struct LayoutDetails {
-    pub variants: Variants,
+    /// Says where the fields are located.
```

Comment:
"are located within the type" could be slightly clearer.

Reply:
Also I don't know if you want to include this but unions and arrays are distinguished here (in call-ABI-significant ways even).

Reply:
> Also I don't know if you want to include this but unions and arrays are distinguished here (in call-ABI-significant ways even).

Well unions and arrays and "everything else" are distinguished here, as one can tell from the `FieldPlacement` variants. Or is there anything special about union vs array that I am missing?

Reply:
> as one can tell from the `FieldPlacement` variants

Sure, I was just wondering if you want to also note something here.

Reply:
I don't see why one would call out specifically arrays and unions, out of all the types one could mention here.